### PR TITLE
Simplesaml refactor

### PR DESCRIPTION
**Pull request by woodseowl**
Moving simplesaml code into a helper + doing some clean up to simplify.

Testing
- [ ] Can log in via user/password and get redirected to the ~~/home page~~ destination
  - [ ] Clicking on "Add a Statement of Need" makes the destination the Add form
  - [ ] Clicking on "Login" makes the destination "/home"
- [ ] Can log in via NetID and get redirected to the ~~/home page~~ destination
  - [ ] Clicking on "Add a Statement of Need" makes the destination the Add form
  - [ ] Clicking on "Login" makes the destination "/home"
- [x] Can log out and get redirected to the welcome page (/)
- [x] Code is understandable and well organized

### CODE REVIEW

File: app/Http/Controllers/Auth/LoginController.php [hunk 1]
```diff
@@ -37,4 +37,20 @@ public function __construct()
...
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function showLoginForm()
+    {
+        $destination = redirect()->intended()->getTargetUrl();
+
+        // Since redirect()->intended() pulls the destination off the stack, we need to push it back.
+        redirect()->setIntendedUrl($destination);
+
```
Comment (philwilliammee) [inline app/Http/Controllers/Auth/LoginController.php:52]: does not redirect to /home but does redirect to destination. This is probably best from user experience. could be changed to `$destination = $this->$redirectTo;`

File: app/Helpers/SimpleSaml.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Helpers;
+
+require_once __DIR__ . '/../../simplesamlphp/lib/_autoload.php';
+
+/**
+ * Class SimpleSaml
+ *
+ * @TODO: Push this into a service container so it doesn't have to initialize multiple times
```
Comment (philwilliammee) [inline app/Helpers/SimpleSaml.php:10]: 👍 good idea

File: public/index.php [hunk 3]
```diff
@@ -9,41 +9,7 @@
...
-    }
-}
-
-$_ENV['SIMPLESAML_LOGIN_URL'] = $auth->getLoginURL();
-$_ENV['SIMPLESAML_LOGOUT_URL'] = $auth->getLogoutURL();
-$_ENV['SIMPLESAML_REMOTE_USERNAME'] = $full_name;
-$_ENV['SIMPLESAML_REMOTE_USER'] = $netid;
-
-// $_ENV['SIMPLESAML_REMOTE_USER_ATTR'] = json_encode($attr, true);
+require_once __DIR__ . '/../simplesamlphp/lib/_autoload.php';
```
Comment (philwilliammee) [inline public/index.php:12]: why is this still required? Just wondering where its being used?

**Comment by philwilliammee** [comment]
see comment https://github.com/CU-CommunityApps/cd-son/pull/37#discussion_r409001542

After login I don't get redirected to home page. But do get redirected to destination. To me either is fine. Its really up to the site owner(shannon) redirect back to destination is probably better in the long run.

File: app/Http/Controllers/Auth/LoginController.php (hunk 1 above)
Comment (woodseowl) [reply app/Http/Controllers/Auth/LoginController.php:52]: I updated the test instructions to reflect the intent of the code. So, I think we agree?

File: public/index.php (hunk 3 above)
Comment (woodseowl) [reply public/index.php:12]: Yeah, the problem is that for reasons unclear to me, simplesamlphp autoload has to load before the composer autoloader or it throws an error. I also tried to overcome the same composer-based simplesamlphp dependency conflicts you encountered but couldn't come up with a way to do it. (I'm more inclined to address #29 and have a SAML2 library that is strictly focused on being an SP.) So, this is the compromise for now.

**Comment by woodseowl** [comment]
Thanks @philwilliammee. I added a couple comments for your questions and will merge it now.

File: app/Http/Controllers/Auth/LoginController.php (hunk 1 above)
Comment (philwilliammee) [reply app/Http/Controllers/Auth/LoginController.php:52]: clicking the login button doesnt redirect me to home. It just takes me back to /

Comment (woodseowl) [reply app/Http/Controllers/Auth/LoginController.php:52]: I'll just keep updating the test instructions until they match the code! That's test driven development, right??

**Pull request merged**